codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -782,11 +783,53 @@ picToObj' pic =
+describePicture (ThickPolygon _ pts w) = printf "thickPolygon { points = %s , width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
+describePicture (SolidRectangle _ w h) = printf "solidRectangle { width = %4f , height = %4f }" w h
+describePicture (ThickRectangle _ lw w h) = printf "thickRectangle { linewidth = %4f , width = %4f , height = %4f }" lw w h
+describePicture (ClosedCurve _ pts) = printf "closedCurve { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])
+describePicture(ThickClosedCurve _ pts w) = printf "thickClosedCurve { points = %s, width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
+describePicture (Circle _ r) = printf "circle { radius = %4f }" r
+describePicture (SolidCircle _ r) = printf "solidCircle { radius = %4f }" r
+describePicture (ThickCircle _ lw r) = printf "thickCircle { linewidth = %4f , radius = %4f }" lw r
+describePicture (Polyline _ pts) = printf "polyline { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])
+describePicture (ThickPolyline _ pts w) = printf "thickPolyline { points = %s, width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
+describePicture (Curve _ pts) = printf "curve { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])
+describePicture (ThickCurve _ pts w) = printf "thickCurve { points = %s, width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
+describePicture (Sector _ b e r) = printf "sector { startAngle = %4f , endAngle = %4f , radius = %4f}" b e r
+describePicture (Arc _ b e r) = printf "arc { startAngle = %4f , endAngle = %4f , radius = %4f}" b e r
+describePicture (ThickArc _ b e r w) = printf "thickArc { startAngle = %4f , endAngle = %4f , radius = %4f , width = %4f}" b e r w
+describePicture (Text _ txt) = printf "text { text = '%s' }" txt
+describePicture (Blank _) = printf "blank"
+describePicture (StyledText _ style font txt) = printf " styledText { style = %s , font = %s , txt = '%s' }" (show style) (show font) txt
+describePicture (Color _ (RGBA r g b a) _) = printf "colored { color = RGBA(%4f, %4f, %4f, %4f) }" r g b a
+describePicture (Translate _ x y _) = printf "translated { x = %4f , y = %4f }" x y
+describePicture (Scale _ x y _) = printf "scaled { x = %4f , y = %4f }" x y
+describePicture (Rotate _ angle _) = printf "rotated { angle = %4f }" angle
+describePicture (Dilate _ k _) = printf "dilated { factor = %4f }" k
+describePicture (Logo _) = printf "codeWorldLogo"
+describePicture (CoordinatePlane _) = printf "coordinatePlane"
+describePicture (Pictures _) = printf "pictures"
+
 setCallInfo :: Picture -> Object -> IO ()
 setCallInfo pic obj =
     case findCSMain (getPictureCS pic) of
-        Just (callName, src) -> do
-            setProp "name" (pToJSVal $ callName) obj
+        Just (_, src) -> do
+            setProp "name" (pToJSVal $ describePicture pic) obj
```

Comment:
Also, don't forget to use the `trim` function down here

Reply:
So I think the function works but I may be missing some edge cases. Can you give me a teeny tiny hint as to where it should go? :D